common/lib/common-utils/src/config.ts

```diff
@@ -0,0 +1,78 @@
+import { Lazy } from "./lazy";
+
+const isLocalStorageAvailable = new Lazy(()=>{
+    const localStorageTestKey = "FluidCommonUtilsLocalStorageTestKey";
+    try {
+        localStorage.setItem(localStorageTestKey, "v");
+        localStorage.removeItem(localStorageTestKey);
+        return true;
+    } catch (e) {
+        return false;
+    }
+});
+
+export function createLocalStorageConfigurationProxy<C extends Record<string, unknown>>(
+    namspace: string, target: C): Readonly<C> {
+    if (isLocalStorageAvailable.value) {
+        return new Proxy<C>(
+            target,
+            new LocalStorageConfigurationProxyHandler<C>(namspace));
+    }
+    return target;
+}
+
+export class LocalStorageConfigurationProxyHandler<T extends Record<string, unknown>> implements ProxyHandler<T> {
+    private readonly configCache = new Map<string, unknown>();
+    constructor(private readonly namespace: string) {
```

Comment:
Think about how something like nconf can be wrap on the node side as well to allow argv/env/json config files to provide options.

Reply:
i was thinking about that exact thing, and i'm not sure. it's an open question to me if we should setup the proxy, like I'm doing here, or ask partners to do it, basically they should wrap it before they give it to us. I see benefits and draw backs to both, so not sure which direction to go